### ISSUE TITLE
Ability to customize binary bundle location

### ISSUE THREAD

**Issue by nevir** [Microsoft/react-native-code-push#45]
I was a bit surprised that [`main.jsbundle` is hard coded](https://github.com/Microsoft/react-native-code-push/blob/master/CodePush.m#L31). `[CodePush setBinaryBundleURL:]` maybe?

**Comment by lostintangent** [comment]
Agreed, we should change this. We did this primarily based on the fact that the current prescription leads you down the path of calling the bundle main.jsbundle (e.g. the xcodeproj and Gradle build tasks generate that file name by default, the default templates reference that name). 

We weren't sure how common it was for folks to change this, so we went with the naïve solution first, but definitely want to be fully accommodating moving forward. Out of curiosity, why did you guys customize your bundle file name?
